### PR TITLE
Add missing longest_side

### LINKED ISSUE #77839: The throwing knives are too small
### Describe the bug

https://github.com/CleverRaven/Cataclysm-DDA/blob/156cf135f3508ad1b8896c1749a45a36c882a976/data/json/items/ranged/throwing.json#L124

The description does not specify length, so the game counts as 100^(1/3) = 4.64 rounded to 5 centimeters.

I don't understand anything about knives and throwing them. So I looked it up on the Internet. I don't know how current this is, but according to The World Knife Throwing League.

https://worldknifethrowingleague.com/best-throwing-knife/

The overall length of the knife ranges from 13.5 inches to 16 inches or 34 to 40 centimeters. These are the knives they sell and most likely use in competitions.

An additional internet search finds smaller knives: about 20 centimeters long

- JXE JXO Throwing Spikes: https://www.amazon.com/JXE-JXO-Throwing-Spikes-Stainless/dp/B0BN11QJMH
9.4 inch ~ 24 cm, 0.44lbs ~ 200 g
- MedievalDepot Midnight Flyer Triple Threat Throwing Knives https://www.amazon.com/MedievalDepot-Midnight-Triple-Threat-Throwing/dp/B0188S17S4
7.5 inch ~ 19 cm

### Attach save file

n/a

### Steps to reproduce

n/a

### Expected behavior

1. Since the game uses an abstract throwing knife, I suggest focusing on the dimensions 20 x 4 x 0.7 cm.
Then its volume will be equal to: 56 ml. And the weight should be reduced to 180-220 grams.
```
    "volume": "56 ml",
    "weight": "200 g",
    "longest_side": "20 cm",
```
2. Use a more accurate model, for example: https://store.worldknifethrowingleague.com/products/the-wktl-merlin-throwing-knife-set-of-3

>  Weight: 1.2lb
    Thickness: 6 mm (0.236")
    Overall Length: 14.25 in
    Blade Width: 1.5 in
    Guard Width: 4.7 in

~ 362 x 38 x 6 mm = ~83mm + some volume on the guard = ~100 ml
```
    "volume": "100 ml",
    "weight": "544 g",
    "longest_side": "362 mm",
```
3. Any other knife of your choice

### Screenshots

_No response_

### Versions and configuration

master

### Additional context

I don't see the point of adding heavy and long throwing knives. You can throw a bayonet too. If the throwing knives are heavier, they will be useless. So it's better to consider relatively light knives and maybe we should lower their damage a bit?

https://github.com/CleverRaven/Cataclysm-DDA/blob/156cf135f3508ad1b8896c1749a45a36c882a976/data/json/items/melee/swords_and_blades.json#L306-L307

Has exactly the same throwing damage as a throwing knife. But it's 55 grams lighter.

### PR DESCRIPTION
**Pull request by xcdxbaxa9**
#### Summary
Balance "Add missing longest_side"

#### Purpose of change
Closes #77839

#### Describe the solution
Added missing longest_side fields, updated volume and weight to more reasonable values.
Used suggestions from linked issue.

#### Describe alternatives you've considered
None.

### CODE REVIEW

**Comment by IdleSol** [comment]
I would like to ask for an additional modification to the throwing knife sheaths.

https://github.com/CleverRaven/Cataclysm-DDA/blob/65b3721c1e4624b323d0bd9d69e8f1bdee5a0183/data/json/items/armor/sheath.json#L115

```diff
--        "max_item_length": "10 cm"
++        "max_item_length": "20 cm"
```
Thank you.

**Comment by RedMisao** [comment]
Regarding the weight change: have you tested the before and after against a target? Weight is used in the throwing formula, and increasing it may break the item.

**Comment by IdleSol** [comment]
>Regarding the weight change: have you tested the before and after against a target? Weight is used in the throwing formula, and increasing it may break the item.

You mean the axe? Isn't thrown_damage used for damage?
https://github.com/CleverRaven/Cataclysm-DDA/blob/45a67eacdd749fedb6d2e0f7baef2ed036366c0f/data/json/items/ranged/throwing.json#L119

**Comment by xcdxbaxa9** [comment]
> Regarding the weight change: have you tested the before and after against a target? Weight is used in the throwing formula, and increasing it may break the item.

Do you mean break as in damage the item? Or break as in make it too powerful or too weak?
I've tested throwing axe before and after, it didn't break. As for effectiveness it didn't look like much changed to me.

**Comment by RedMisao** [comment]
> You mean the axe? Isn't `thrown_damage` used for damage?

Yes, that is used to calculate damage. I'm referring to the distance and accuracy of the thrown item. Items A heavier than B, is less accurate, and seems to "glance" more (although I'm not sure about this, could be due being less accurate).


> Do you mean break as in damage the item? Or break as in make it too powerful or too weak? I've tested throwing axe before and after, it didn't break. As for effectiveness it didn't look like much changed to me.

As in make it too weak for use as a throwing weapon. If you tested it and found no difference then nvm.

File: data/json/items/ranged/throwing.json [hunk 1]
```diff
@@ -131,8 +132,9 @@
     "color": "light_gray",
     "description": "A thin, flat knife made for throwing.  Its ineffective cutting edge and odd shape make it unsuitable for use as a tool.",
     "material": [ "steel" ],
-    "volume": "100 ml",
-    "weight": "460 g",
+    "volume": "56 ml",
+    "weight": "200 g",
+    "longest_side": "20 cm",
```
Comment (Maleclypse) [inline data/json/items/ranged/throwing.json:137]:
```suggestion
    "longest_side": "35 cm",
```
World Knife Throwing League (WKTL) requires that they are between 13.5 and 16 inches long. This length allows throwers of different heights and arm lengths to hit targets at distances of 10 and 15 feet.

File: data/json/items/armor/sheath.json [hunk 2]
```diff
@@ -177,7 +177,7 @@
         "moves": 30,
         "max_contains_volume": "100 ml",
         "max_contains_weight": "500 g",
-        "max_item_length": "10 cm"
+        "max_item_length": "20 cm"
```
Comment (Maleclypse) [inline data/json/items/armor/sheath.json:180]:
```suggestion
        "max_item_length": "35 cm"
```

File: data/json/items/armor/sheath.json [hunk 3]
```diff
@@ -168,7 +168,7 @@
...
```
Comment (Maleclypse) [inline data/json/items/armor/sheath.json:171]:
```suggestion
        "max_item_length": "35 cm"
```

File: data/json/items/armor/sheath.json [hunk 4]
```diff
@@ -159,7 +159,7 @@
...
```
Comment (Maleclypse) [inline data/json/items/armor/sheath.json:162]:
```suggestion
        "max_item_length": "35 cm"
```

File: data/json/items/armor/sheath.json [hunk 5]
```diff
@@ -150,7 +150,7 @@
...
```
Comment (Maleclypse) [inline data/json/items/armor/sheath.json:153]:
```suggestion
        "max_item_length": "35 cm"
```

File: data/json/items/armor/sheath.json [hunk 6]
```diff
@@ -132,7 +132,7 @@
...
```
Comment (Maleclypse) [inline data/json/items/armor/sheath.json:135]:
```suggestion
        "max_item_length": "35 cm"
```

File: data/json/items/armor/sheath.json [hunk 7]
```diff
@@ -141,7 +141,7 @@
...
```
Comment (Maleclypse) [inline data/json/items/armor/sheath.json:144]:
```suggestion
        "max_item_length": "35 cm"
```

File: data/json/items/armor/sheath.json (hunk 6 above)
Comment (Maleclypse) [reply data/json/items/armor/sheath.json:135]: Change `made to hold 6 very small knives` to `made to hold 6 very thin knives`

**Review by Maleclypse — CHANGES_REQUESTED** [review]
Thanks for doing this!

**Comment by IdleSol** [comment]
@Maleclypse

If you increase the length, you have to increase the weight (~500g). And a throwing knife becomes a copy of a combat knife. Only worse.

USMC bayonet:
https://github.com/CleverRaven/Cataclysm-DDA/blob/156cf135f3508ad1b8896c1749a45a36c882a976/data/json/items/melee/swords_and_blades.json#L306-L307

**Comment by Maleclypse** [comment]
> @Maleclypse
> 
> If you increase the length, you have to increase the weight (~500g). And a throwing knife becomes a copy of a combat knife. Only worse.
> 
> USMC bayonet:
> 
> https://github.com/CleverRaven/Cataclysm-DDA/blob/156cf135f3508ad1b8896c1749a45a36c882a976/data/json/items/melee/swords_and_blades.json#L306-L308

I came up with 340 grams for a 35 cm throwing knife.  But if our system is unable to differentiate between made to be thrown items versus not designed for throwing I suppose these are the limits of the system.

**Comment by xcdxbaxa9** [comment]
> > @Maleclypse
> > If you increase the length, you have to increase the weight (~500g). And a throwing knife becomes a copy of a combat knife. Only worse.
> > USMC bayonet:
> > https://github.com/CleverRaven/Cataclysm-DDA/blob/156cf135f3508ad1b8896c1749a45a36c882a976/data/json/items/melee/swords_and_blades.json#L306-L308
> 
> I came up with 340 grams for a 35 cm throwing knife. But if our system is unable to differentiate between made to be thrown items versus not designed for throwing I suppose these are the limits of the system.

Can you elaborate on how you came up with 340 grams for 35 cm throwing knife? If we assume that average density of stainless steel is 7.85 g/cm^3 and volume remains the same (56 ml,) then weight gets around 450 grams, give or take.

**Comment by Maleclypse** [comment]
> > > @Maleclypse
> > > If you increase the length, you have to increase the weight (~500g). And a throwing knife becomes a copy of a combat knife. Only worse.
> > > USMC bayonet:
> > > https://github.com/CleverRaven/Cataclysm-DDA/blob/156cf135f3508ad1b8896c1749a45a36c882a976/data/json/items/melee/swords_and_blades.json#L306-L308
> > 
> > 
> > I came up with 340 grams for a 35 cm throwing knife. But if our system is unable to differentiate between made to be thrown items versus not designed for throwing I suppose these are the limits of the system.
> 
> Can you elaborate on how you came up with 340 grams for 35 cm throwing knife? If we assume that average density of stainless steel is 7.85 g/cm^3 and volume remains the same (56 ml,) then weight gets around 450 grams, give or take.
> 
> upd: Tested 200 g and 450 g throwing knifes. On average 200 g variants were dealing around 14 points of damage to regular zombies and 450 g variants were dealing somewhere in the range of 20-25 points of damage. I don't know if it makes them too overpowered though.

Volume could change.  Our volumes are generally overstated for objects because it's not displacement volume but a rough estimate of how much volume a thing would take jumbled in a bag.  So for items with any portrusions it's different.  Either way I'm not super interested in this so I'm not pushing my view on you.  If you tell me this is right I'm going to believe you compared to my quick googling that could have been pulling details from multiple different sources in inappropriate ways.

**Comment by IdleSol** [comment]
#77839 :
>https://worldknifethrowingleague.com/best-throwing-knife/
>
>The overall length of the knife ranges from 13.5 inches to 16 inches or 34 to 40 centimeters. These are the knives they sell and most likely use in competitions.
>
>An additional internet search finds smaller knives: about 20 centimeters long
>
>- JXE JXO Throwing Spikes: https://www.amazon.com/JXE-JXO-Throwing-Spikes-Stainless/dp/B0BN11QJMH
9.4 inch ~ 24 cm, 0.44lbs ~ 200 g
>- MedievalDepot Midnight Flyer Triple Threat Throwing Knives https://www.amazon.com/MedievalDepot-Midnight-Triple-Threat-Throwing/dp/B0188S17S4
7.5 inch ~ 19 cm

**Comment by xcdxbaxa9** [comment]
I've left the weight unchanged as in my opinion then throwing knives become too strong.

**Pull request merged**